department-of-veterans-affairs/abd-vro
# MCP-2262 Use new PDF endpoint

## What was the problem?

Instead of using two API calls, one to generate and one to fetch the PDF, use the new API that generates and fetches in one go.

Furthermore, we want to store evidence summary and assessment results in the DB

Associated tickets or Slack threads:

- [MCP-2262](https://amida.atlassian.net/browse/MCP-2262)

## How does this fix it?[^1]

- Call the new PDF endpoint that generates and returns the PDF payload in one go
- Persist the evidence summary before calling "generate PDF"
- added some more details in the audit event trail.

## How to test this PR
- The functionality should be the same 
- The audits for a claim that is not offramped should be something like this:

Checking if claim is ready...
Started claim processing.
Collecting evidence
There is insufficient evidence. Ordering an exam
Generating PDF
Uploaded PDF
Removing Special Issue
Claim satisfies TSOJ condition. Updated status.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
